### ISSUE TITLE
Updating to CMake 3.30.3 causes Segmentation fault

### ISSUE THREAD

**Issue by Lablace** [conda-forge/cmake-feedstock#222]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

## Problem

After upgrading conda-forge packaged cmake from version 3.30.2 to 3.30.3, a simple call like `cmake --version` causes Segmentation fault. If I reinstall the 3.30.2, everything goes normally again. Now I have froze my daily update routine and stick to 3.30.2 temporarily.

## System

Ubuntu 22.04.1 LTS 

## Linked packages

```plaintext
$> ldd cmake
linux-vdso.so.1 (0x00007ffdf45b2000)
libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fefc1caf000)
libcurl.so.4 => MAMBA_ENV_DIR/bin/../lib/libcurl.so.4 (0x00007fefc1be2000)
libbz2.so.1.0 => MAMBA_ENV_DIR/bin/../lib/libbz2.so.1.0 (0x00007fefc1bce000)
liblzma.so.5 => MAMBA_ENV_DIR/bin/../lib/liblzma.so.5 (0x00007fefc1ba5000)
libzstd.so.1 => MAMBA_ENV_DIR/bin/../lib/libzstd.so.1 (0x00007fefc1a91000)
librhash.so.1 => MAMBA_ENV_DIR/bin/../lib/librhash.so.1 (0x00007fefc1a5b000)
libuv.so.1 => MAMBA_ENV_DIR/bin/../lib/libuv.so.1 (0x00007fefc1a2b000)
libz.so.1 => MAMBA_ENV_DIR/bin/../lib/libz.so.1 (0x00007fefc1a10000)
libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fefc1a0b000)
libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fefc17df000)
libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fefc16f8000)
libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fefc16d6000)
libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fefc14ae000)
/lib64/ld-linux-x86-64.so.2 (0x00007fefc2904000)
libnghttp2.so.14 => MAMBA_ENV_DIR/bin/../lib/./libnghttp2.so.14 (0x00007fefc1482000)
libssh2.so.1 => MAMBA_ENV_DIR/bin/../lib/./libssh2.so.1 (0x00007fefc143d000)
libssl.so.3 => MAMBA_ENV_DIR/bin/../lib/./libssl.so.3 (0x00007fefc134d000)
libcrypto.so.3 => MAMBA_ENV_DIR/bin/../lib/./libcrypto.so.3 (0x00007fefc0e0b000)
libgssapi_krb5.so.2 => MAMBA_ENV_DIR/bin/../lib/./libgssapi_krb5.so.2 (0x00007fefc0db8000)
librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fefc0db3000)
libkrb5.so.3 => MAMBA_ENV_DIR/bin/../lib/././libkrb5.so.3 (0x00007fefc0cdd000)
libk5crypto.so.3 => MAMBA_ENV_DIR/bin/../lib/././libk5crypto.so.3 (0x00007fefc0cc3000)
libcom_err.so.3 => MAMBA_ENV_DIR/bin/../lib/././libcom_err.so.3 (0x00007fefc0cbd000)
libkrb5support.so.0 => MAMBA_ENV_DIR/bin/../lib/././libkrb5support.so.0 (0x00007fefc0caf000)
libkeyutils.so.1 => MAMBA_ENV_DIR/bin/../lib/././libkeyutils.so.1 (0x00007fefc0ca8000)
libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007fefc0c94000)
```

## GDB output

```plaintext
$> gdb cmake corefile
(gdb) bt
#0  0x00007f2d68282f8a in std::ostream::sentry::sentry(std::ostream&) () from /lib/x86_64-linux-gnu/libstdc++.so.6
#1  0x00007f2d68283a0c in std::basic_ostream<char, std::char_traits<char> >& std::__ostream_insert<char, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&, char const*, long) ()
   from /lib/x86_64-linux-gnu/libstdc++.so.6
#2  0x0000556ceb61f5ff in cmDocumentation::PrintVersion(std::ostream&) ()
#3  0x0000556ceb628d96 in cmDocumentation::PrintRequestedDocumentation(std::ostream&) ()
#4  0x0000556ceb5fc3e4 in (anonymous namespace)::do_cmake(int, char const* const*) ()
#5  0x0000556ceb5d3cc0 in main ()
```
 

### Installed packages

```shell
Name                Version       Build                  Channel
────────────────────────────────────────────────────────────────────────
  _go_select          2.3.0         cgo                    conda-forge
  _libgcc_mutex       0.1           conda_forge            conda-forge
  _openmp_mutex       4.5           2_gnu                  conda-forge
  bottle              0.12.23       pyhd8ed1ab_0           conda-forge
  brotli-python       1.1.0         py312h2ec8cdc_2        conda-forge
  bzip2               1.0.8         h4bc722e_7             conda-forge
  c-ares              1.33.1        heb4867d_0             conda-forge
  ca-certificates     2024.8.30     hbcca054_0             conda-forge
  certifi             2024.8.30     pyhd8ed1ab_0           conda-forge
  charset-normalizer  3.3.2         pyhd8ed1ab_0           conda-forge
  clang-format        18.1.8        default_hf981a13_4     conda-forge
  clang-format-18     18.1.8        default_hf981a13_4     conda-forge
  clang-tools         18.1.8        default_hf981a13_4     conda-forge
  cmake               3.30.3        hf9cb763_0             conda-forge
  colorama            0.4.6         pyhd8ed1ab_0           conda-forge
  conan               2.6.0         pyhd8ed1ab_0           conda-forge
  curl                8.9.1         h18eb788_0             conda-forge
  distro              1.8.0         pyhd8ed1ab_0           conda-forge
  doxygen             1.10.0        h661eb56_0             conda-forge
  fasteners           0.17.3        pyhd8ed1ab_0           conda-forge
  gettext             0.22.5        he02047a_3             conda-forge
  gettext-tools       0.22.5        he02047a_3             conda-forge
  git                 2.46.0        pl5321hb5640b7_0       conda-forge
  go                  1.23.0        hc4c73a6_1             conda-forge
  go-shfmt            3.9.0         ha8f183a_0             conda-forge
  hyperfine           1.18.0        he8a937b_0             conda-forge
  icu                 75.1          he02047a_0             conda-forge
  idna                3.8           pyhd8ed1ab_0           conda-forge
  jinja2              3.1.4         pyhd8ed1ab_0           conda-forge
  jq                  1.7.1         hd590300_0             conda-forge
  keyutils            1.6.1         h166bdaf_0             conda-forge
  krb5                1.21.3        h659f571_0             conda-forge
  ld_impl_linux-64    2.40          hf3520f5_7             conda-forge
  libasprintf         0.22.5        he8f35ee_3             conda-forge
  libasprintf-devel   0.22.5        he8f35ee_3             conda-forge
  libclang-cpp18.1    18.1.8        default_hf981a13_4     conda-forge
  libclang13          18.1.8        default_h9def88c_4     conda-forge
  libcurl             8.9.1         hdb1bdb2_0             conda-forge
  libedit             3.1.20191231  he28a2e2_2             conda-forge
  libev               4.33          hd590300_2             conda-forge
  libexpat            2.6.2         h59595ed_0             conda-forge
  libffi              3.4.2         h7f98852_5             conda-forge
  libgcc              14.1.0        h77fa898_1             conda-forge
  libgcc-ng           14.1.0        h69a702a_1             conda-forge
  libgettextpo        0.22.5        he02047a_3             conda-forge
  libgettextpo-devel  0.22.5        he02047a_3             conda-forge
  libgfortran         14.1.0        h69a702a_1             conda-forge
  libgfortran-ng      14.1.0        h69a702a_1             conda-forge
  libgfortran5        14.1.0        hc5f4f2c_1             conda-forge
  libgomp             14.1.0        h77fa898_1             conda-forge
  libiconv            1.17          hd590300_2             conda-forge
  libidn2             2.3.7         hd590300_0             conda-forge
  libllvm18           18.1.8        h8b73ec9_2             conda-forge
  libnghttp2          1.58.0        h47da74e_1             conda-forge
  libnsl              2.0.1         hd590300_0             conda-forge
  libsqlite           3.46.1        hadc24fc_0             conda-forge
  libssh2             1.11.0        h0841786_0             conda-forge
  libstdcxx           14.1.0        hc0a3c3a_1             conda-forge
  libstdcxx-ng        14.1.0        h4852527_1             conda-forge
  libunistring        0.9.10        h7f98852_0             conda-forge
  libuuid             2.38.1        h0b41bf4_0             conda-forge
  libuv               1.48.0        hd590300_0             conda-forge
  libxcb              1.16          hb9d3cd8_1             conda-forge
  libxcrypt           4.4.36        hd590300_1             conda-forge
  libxml2             2.12.7        he7c6b58_4             conda-forge
  libzlib             1.3.1         h4ab18f5_1             conda-forge
  markupsafe          2.1.5         py312h66e93f0_1        conda-forge
  meson               1.5.1         pyhd8ed1ab_1           conda-forge
  ncurses             6.5           he02047a_1             conda-forge
  ninja               1.12.1        h297d8ca_0             conda-forge
  oniguruma           6.9.9         hd590300_0             conda-forge
  openssl             3.3.2         hb9d3cd8_0             conda-forge
  p7zip               16.02         h9c3ff4c_1001          conda-forge
  patch-ng            1.17.4        pyh9f0ad1d_0           conda-forge
  pcre2               10.44         hba22ea6_2             conda-forge
  perl                5.32.1        7_hd590300_perl5       conda-forge
  pip                 24.2          pyh8b19718_1           conda-forge
  pkg-config          0.29.2        h4bc722e_1009          conda-forge
  pluginbase          1.0.1         pyhd8ed1ab_0           conda-forge
  pthread-stubs       0.4           h36c2ea0_1001          conda-forge
  pyjwt               2.9.0         pyhd8ed1ab_1           conda-forge
  pysocks             1.7.1         pyha2e5f31_6           conda-forge
  python              3.12.5        h2ad013b_0_cpython     conda-forge
  python-dateutil     2.9.0         pyhd8ed1ab_0           conda-forge
  python_abi          3.12          5_cp312                conda-forge
  pyyaml              6.0.2         py312h66e93f0_1        conda-forge
  readline            8.2           h8228510_1             conda-forge
  requests            2.32.3        pyhd8ed1ab_0           conda-forge
  rhash               1.4.4         hd590300_0             conda-forge
  ruff                0.6.2         py312h78ba408_1        conda-forge
  setuptools          73.0.1        pyhd8ed1ab_0           conda-forge
  six                 1.16.0        pyh6c4a22f_0           conda-forge
  starship            1.20.1        he0f44a0_1             conda-forge
  taplo               0.9.3         h1de38c7_0             conda-forge
  tk                  8.6.13        noxft_h4845f30_101     conda-forge
  tzdata              2024a         h8827d51_1             conda-forge
  urllib3             1.26.19       pyhd8ed1ab_0           conda-forge
  vim                 9.1.0611      py312pl5321h419f9bc_1  conda-forge
  wget                1.21.4        hda4d442_0             conda-forge
  wheel               0.44.0        pyhd8ed1ab_0           conda-forge
  xorg-kbproto        1.0.7         h7f98852_1002          conda-forge
  xorg-libice         1.1.1         hd590300_0             conda-forge
  xorg-libsm          1.2.4         h7391055_0             conda-forge
  xorg-libx11         1.8.9         hb711507_1             conda-forge
  xorg-libxau         1.0.11        hd590300_0             conda-forge
  xorg-libxdmcp       1.1.3         h7f98852_0             conda-forge
  xorg-libxt          1.3.0         hd590300_1             conda-forge
  xorg-xextproto      7.3.0         h0b41bf4_1003          conda-forge
  xorg-xproto         7.0.31        h7f98852_1007          conda-forge
  xxhash              0.8.2         hd590300_0             conda-forge
  xz                  5.2.6         h166bdaf_0             conda-forge
  yaml                0.2.5         h7f98852_2             conda-forge
  zlib                1.3.1         h4ab18f5_1             conda-forge
  zstd                1.5.6         ha6fb4c9_0             conda-forge
```

### Environment info

```shell
libmamba version : 1.5.9
     micromamba version : 1.5.9
           curl version : libcurl/8.9.1 OpenSSL/3.3.1 zlib/1.3.1 zstd/1.5.6 libssh2/1.11.0 nghttp2/1.58.0
     libarchive version : libarchive 3.7.4 zlib/1.2.13 bz2lib/1.0.8 libzstd/1.5.6
       envs directories : MAMBA_ROOT_PREFIX/envs
          package cache : MAMBA_ROOT_PREFIX/pkgs
                          HOME/.mamba/pkgs
            environment : MAMBA_ENV_NAME
           env location : MAMBA_ENV_DIR
      user config files : HOME/.mambarc
 populated config files : HOME/.config/mamba/.mambarc
       virtual packages : __unix=0=0
                          __linux=5.15.0=0
                          __glibc=2.35=0
                          __archspec=1=x86_64-v3
       base environment : MAMBA_ROOT_PREFIX
               platform : linux-64
```

**Comment by Lablace** [comment]
I think it is not a bug related to cmake itself, as on my Mac the brew packaged cmake 3.30.3 works normally.

**Comment by Lablace** [comment]
I solved this as a bonus while managing another environment, the problem seems to be with the `libstdc++.so`. It had to be a **newer** version which `micromamba` failed (or forgot, perhaps) to install for me. The same problem exists for newer version of `libarrow`, which requires that `so` to support`GLIBCXX_3.4.32`, while the one on my system at most supports only `GLIBCXX_3.4.30`.

I then deleted the whole environment and reinstalled all packages, at this time, a newer `so` was installed in the environment's library directory where `libarrow` can find and link to it. I then updated my `LD_LIBRARY_PATH` to let `ld` search that library path, and `cmake` worked like a charm (without segfault).

So to sum up, **there is still some problem for `micromamba` to detect whether a packaged `libstdc++.so` is needed to be installed.** It seems to have that (maybe partial?) check to some extent, but the first check result seems to block or outweigh successive checks (if any).

I don't know if you ever check these issues, as the update and release procedures seem to be quite streamlined. But thanks very much for providing these packages and hope it can get improved.

**Comment by Lablace** [comment]
> > So to sum up, **there is still some problem for `micromamba` to detect whether a packaged `libstdc++.so` is needed to be installed.**
> 
> If you have a way to reproduce or at least describe the problem, please open an issue on the mamba/micromamba feedstock, or directly in https://github.com/mamba-org/mamba

Thanks. I've tried to reproduce during the last weekend but in vain. I just deleted the old environment without exporting the package info and as a result I cannot find out what packages were there in problem. I tried to rollback them to match the older `cmake` version, but this time it just shipped with the older `libstdc++.so`, and after upgrading the `so` got updated as well. This is kind of weird, I will try again when I am free.

**Comment by h-vetinari** [comment]
IMO you should assume that the problem was fixed by the new mamba release, and not waste more time on this unless it actually happens again

**Comment by Lablace** [comment]
😆 you are right, let it go

**Comment by isuruf** [comment]
It was probably https://github.com/mamba-org/mamba/issues/3423 which was fixed.

**Comment by Lablace** [comment]
Thanks, it seems to be that. I should have checked the issues myself 😆.
